### PR TITLE
examples: NNS update, part 1

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Port https://github.com/neo-project/non-native-contracts/pull/23.

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
https://github.com/neo-project/non-native-contracts/pull/23 is linked.

File: examples/nft-nd-nns/nns.go [hunk 1]
```diff
@@ -47,8 +47,8 @@ const (
 	maxRegisterPrice = 1_0000_0000_0000
 	// maxRootLength is the maximum domain root length.
 	maxRootLength = 16
-	// maxDomainNameFragmentLength is the maximum length of the domain name fragment.
-	maxDomainNameFragmentLength = 62
+	// maxDomainNameFragmentLength is the maximum length of the domain name fragment
```
Comment (roman-khimov) [inline examples/nft-nd-nns/nns.go:50]: What has happened to the dot?

File: examples/nft-nd-nns/nns_test.go [hunk 2]
```diff
@@ -154,9 +157,16 @@ func TestRegisterAndRenew(t *testing.T) {
...
+	for i := 0; i < maxDomainNameFragmentLength; i++ {
+		maxLenFragment += "q"
+	}
+	c.Invoke(t, true, "isAvailable", maxLenFragment+".com")
+	c.Invoke(t, true, "register", maxLenFragment+".com", e.CommitteeHash)
+	c.InvokeFail(t, "invalid domain name format", "register", maxLenFragment+"q.com", e.CommitteeHash)
 
 	c.Invoke(t, true, "isAvailable", "neo.com")
-	c.Invoke(t, 0, "balanceOf", e.CommitteeHash)
+	c.Invoke(t, 1, "balanceOf", e.CommitteeHash)
```
Comment (roman-khimov) [inline examples/nft-nd-nns/nns_test.go]: Does this belong to bd3722041a43fb9ac85c79517e903472a3d27cd7?

**Review by roman-khimov — COMMENTED** [review]
It doesn't seem to be equivalent to neo-project/non-native-contracts#23, can we only have here things that are a part of it? TLD/admin/expiration checks are all from different patch sets.

File: examples/nft-nd-nns/nns.go (hunk 1 above)
Comment (AnnaShaleva) [reply examples/nft-nd-nns/nns.go]: Fixed.

File: examples/nft-nd-nns/nns_test.go (hunk 2 above)
Comment (AnnaShaleva) [reply examples/nft-nd-nns/nns_test.go:169]: Definitely.

**Comment by AnnaShaleva** [comment]
> TLD/admin/expiration checks are all from different patch sets.

For C# set of PRs I've changed the order of TLD/expiration commits which results in a painful rebase, because these commits are tightly related. Am I supposed to perform the same thing for us?

**Comment by AnnaShaleva** [comment]
Ready for review.

**Comment by roman-khimov** [comment]
TLD is _very_ invasive, we need to wait for C# PRs on this then.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**